future/src/test/java/com/spotify/hamcrest/future/ExceptionallyCompletedBlockingCompletionStageTest.java

```diff
@@ -85,4 +88,34 @@ public void testInterruptedMismatchFormatting() throws Exception {
       Thread.interrupted();
     }
   }
+
+  @Test
+  public void testInterruptActuallyInterruptsAssert() throws Exception {
+    final CountDownLatch beforeAssertLatch = new CountDownLatch(1);
+    final CountDownLatch afterAssertLatch = new CountDownLatch(1);
+    final Thread threadToInterrupt = new Thread(() -> {
+      try {
+        beforeAssertLatch.countDown();
+        assertThat(new CompletableFuture<String>(),
+            SUT);
+      } finally {
+        afterAssertLatch.countDown();
+      }
+    });
+    try {
+      threadToInterrupt.setDaemon(true);
+      threadToInterrupt.start();
+
+      beforeAssertLatch.await();
+      TimeUnit.SECONDS.sleep(1);
+      threadToInterrupt.interrupt();
+
+      assertTrue("Failed to interrupt assertion",
+          afterAssertLatch.await(2, TimeUnit.SECONDS));
+    } finally {
+      // Hamcrest goes into blocking method twice,
+      // second interrupt should release it and allow thread to finish:
+      threadToInterrupt.interrupt();
+    }
+  }
```

Comment:
Do you want to verify that the thread dies as well without throwing exceptions? I think that a  `threadToInterrupt.join()` might be in place here.

Reply:
No, I think handling failed invocation is responsibility of junit, we do not need to check it.

Reply:
I meant that by default, if an exception happens on a different thread than the main one that runs the test code, it won't be surfaced and the test won't fail.